### ISSUE TITLE
tf.math.reduce_prod followed by a rank-increasing broadcast_to returns a value scaled by an integer factor under XLA

### ISSUE THREAD

**Issue by laolvfan** [tensorflow/tensorflow#126892]
### Issue type
Bug

### Have you reproduced the bug with TensorFlow Nightly?
Yes — `2.22.0-dev20260905`

### Source
binary

### TensorFlow version
`2.22.0-dev20260905` and `2.22.0-dev20260830`, both CPU

### OS platform and distribution
Linux x86-64

### Python version
3.11

### Current behavior?

`tf.math.reduce_prod` along one axis, followed by `tf.broadcast_to` to a shape
with **one more dimension** than the reduced result (so the whole reduced
sub-tensor gets replicated along a newly-inserted leading axis), returns a
value under `jit_compile=True` that is consistently a small integer multiple
(2x or 4x, depending on the exact shapes) of the correct eager value.

Both results are deterministic and reproducible with a fixed seed — this is
not a numerical-precision or parallel-execution-race artifact: forcing
single-threaded execution (`set_intra_op_parallelism_threads(1)` /
`set_inter_op_parallelism_threads(1)`) does not change the wrong XLA value at
all. A standalone `tf.math.reduce_prod` call on the same kind of data (without
the surrounding Conv2D/AveragePooling2D/broadcast_to context) computes the
same value in both eager and XLA, so the bug depends on this specific op
sequence rather than being a general `reduce_prod` defect.

### Standalone code to reproduce the issue

```python
import tensorflow as tf
import numpy as np

class Model(tf.keras.Model):
    def __init__(self):
        super().__init__()
        self.conv1 = tf.keras.layers.Conv2D(
            filters=64, kernel_size=[8, 8], strides=[8, 8], padding='SAME',
            use_bias=False, name='conv1')

    def call(self, x1):
        x2 = self.conv1(x1)
        pooled = tf.keras.layers.AveragePooling2D(pool_size=2, strides=2, padding='same')(x2)
        reduced = tf.math.reduce_prod(pooled, 1)          # shape (32, 1, 64)
        return tf.broadcast_to(reduced, [32, 32, 1, 64])  # one extra leading dim

tf.random.set_seed(420)
m = Model()
x1 = tf.random.normal(shape=[32, 32, 4, 1])

eager = np.array(m(x1)).ravel()[:3]
xla = np.array(tf.function(lambda a: m(a), jit_compile=True)(x1)).ravel()[:3]

print("eager:", eager)
print("xla:  ", xla)
print("ratio:", xla / eager)
```

Observed output:

```
eager: [-0.01481274 -0.00130865 -0.00499152]
xla:   [-0.05925096 -0.00523458 -0.01996606]
ratio: [4.0000005 3.999998  4.0000005]
```

### Relevant log output

None — no error or warning is emitted.
